api/provider/aws/common.go

```diff
@@ -53,6 +54,26 @@ func describeTaskDefinition(ecsapi ecsiface.ECSAPI, taskDefinitionARN string) (*
 	return output.TaskDefinition, nil
 }
 
+func getLaunchTypeFromEnvironmentID(store tag.Store, environmentID string) (string, error) {
+	tags, err := store.SelectByTypeAndID("environment", environmentID)
+	if err != nil {
+		return "", err
+	}
+
+	if tag, ok := tags.WithKey("type").First(); ok {
+		switch tag.Value {
+		case models.EnvironmentTypeDynamic:
+			return ecs.LaunchTypeFargate, nil
+		case models.EnvironmentTypeStatic:
+			return ecs.LaunchTypeEc2, nil
+		default:
+			return "", fmt.Errorf("Could not find instance launch type for environment '%s'", environmentID)
```

Comment:
[nit] would be useful to show what the tag is, e.g. `"Environment '%s' has invalid 'type' tag: %s" environmentID, tag.Value)`

Reply:
✅